### DISCUSSION TITLE
Shipping Method IDs in ShippingMethodType is different to shippingMethod.ID returned in `createCheckout` or when querying `checkouts`

### DISCUSSION THREAD

**Discussion by byronrode** [saleor/saleor#10940] [Q&A]
In Saleor 2.x we used the following query to return a list of shipping methods. This was done on app load to prevent having to run multiple queries during checkout, as we require low-data usage for our business use case. I.e. our app is often used out in the field and with limited access to data/wifi so we stored the shipping methods in cache to ensure that we are optimising our data usage.

```
query {
      shippingZones(first:$first, channel: "default-channel"){
        edges{
          node {
            id
            name
            shippingMethods {
              id
              name
                price {
                  amount
                }
            }
            warehouses {
              name
            }
          }
        }
      }
    }
```

This used to return the available shipping methods, with the correct ID related to the shipping method. Since updating to 3.x we have had to modify the query as the price is no longer returned, and the ID no longer matches what is required for the `checkoutShippingMethodUpdate` mutation.

We now use the following query:

```
query {
      shippingZones(first:$first, channel: "default-channel"){
        edges{
          node {
            id
            name
            shippingMethods {
              id
              name
              channelListings {
                price {
                  amount
                }
              }
            }
            warehouses {
              name
            }
          }
        }
      }
    }
```

However ShippingMethodType returns a completely different ID than what `shippingMethods` does on the `checkouts` query or in the returned data from `checkoutCreate`. How can we efficiently query and store shipping methods as a single transaction on app startup, rather than having to query the checkouts, or return the results from `checkoutCreate` as this creates additional logic and seems counter-intuitive. 

I was going to open a bug report, but I am not sure it is a bug, and is by design, but that we may just need to change some of our queries.

**Comment by byronrode** [comment]
On further inspection, we can use the following: 

```
{
  shop {
    availableShippingMethods(channel: "default-channel") {
      id
      name
      price {
        amount
      }
    }
  }
}
```

However, while this provides the correct IDs, it does not allow us to filter by shipping zone, and we currently don't have plans to add postal codes etc (as it does not fit our requirements).

**Reply by byronrode** [reply]
And filtering by city with `address` does not have any effect, and the only potential solution I can see is a workaround to set a postal code for each zone and then update our apps to use those to filter, however that, again seems counter-productive.

Surely we should be able to filter available shipping methods by the zones that we have setup, or have the shipping method type ID return the same ID as the above `availableShippingMethods`

**Comment by timuric** [comment]
Hey @byronrode sorry, somehow this thread was neglected, have you managed to find a solution?
